testsuite/MDAnalysisTests/topology/test_fetch_pdb.py

```diff
@@ -0,0 +1,161 @@
+# -*- Mode: python; tab-width: 4; indent-tabs-mode:nil; coding:utf-8 -*-
+# vim: tabstop=4 expandtab shiftwidth=4 softtabstop=4 fileencoding=utf-8
+#
+# MDAnalysis --- https://www.mdanalysis.org
+# Copyright (c) 2006-2017 The MDAnalysis Development Team and contributors
+# (see the file AUTHORS for the full list of names)
+#
+# Released under the Lesser GNU Public Licence, v2.1 or any higher version
+#
+# Please cite your use of MDAnalysis in published work:
+#
+# R. J. Gowers, M. Linke, J. Barnoud, T. J. E. Reddy, M. N. Melo, S. L. Seyler,
+# D. L. Dotson, J. Domanski, S. Buchoux, I. M. Kenney, and O. Beckstein.
+# MDAnalysis: A Python package for the rapid analysis of molecular dynamics
+# simulations. In S. Benthall and S. Rostrup editors, Proceedings of the 15th
+# Python in Science Conference, pages 102-109, Austin, TX, 2016. SciPy.
+# doi: 10.25080/majora-629e541a-00e
+#
+# N. Michaud-Agrawal, E. J. Denning, T. B. Woolf, and O. Beckstein.
+# MDAnalysis: A Toolkit for the Analysis of Molecular Dynamics Simulations.
+# J. Comput. Chem. 32 (2011), 2319--2327, doi:10.1002/jcc.21787
+#
+
+import pytest
+
+import MDAnalysis as mda
+from MDAnalysis.topology.PDBParser import (
+    DEFAULT_CACHE_NAME_DOWNLOADER,
+    HAS_POOCH,
+    SUPPORTED_FILE_FORMATS_DOWNLOADER,
+)
+
+import re
+from urllib import request
+from shutil import rmtree
+from pathlib import Path
+
+if HAS_POOCH:
+    from requests.exceptions import HTTPError
+    import pooch
+
+try:
+    request.urlopen("https://files.wwpdb.org/", timeout=2)
+    HAS_ACCESS_TO_WWPDB = True
+except request.URLError:
+    HAS_ACCESS_TO_WWPDB = False
+
+
+def true_basename(path):
+    """This is needed because pathlib.Path(foo.pdb.gz).stem as foo.pdb"""
+    return Path(path).stem.split(".")[0]
+
+
+@pytest.mark.skipif(
+    HAS_POOCH,
+    reason="Pooch is installed.",
+)
+def test_pooch_installation(tmp_path):
+    with pytest.raises(
+        ModuleNotFoundError,
+        match="pooch is needed as a dependency for fetch_pdb()",
+    ):
+        mda.fetch_pdb("1AKE", cache_path=tmp_path, file_format="cif")
+
+
+@pytest.mark.skipif(not HAS_POOCH, reason="Pooch is not installed.")
+@pytest.mark.skipif(
+    not HAS_ACCESS_TO_WWPDB,
+    reason="Can not connect to https://files.wwpdb.org/",
+)
+class TestDocstringExamples:
+    """This class tests all the examples found in fetch_pdb's docstring"""
+
+    @pytest.mark.parametrize("pdb_id", ["1AKE", "4BWZ"])
+    def test_one_file_download(self, tmp_path, pdb_id):
+        path = mda.fetch_pdb(pdb_id, cache_path=tmp_path, file_format="cif")
+        assert isinstance(path, str)
+        assert true_basename(path) == pdb_id
```

Comment:
```suggestion
        assert path == f"{pdb_id}.cif"
```
It would be better to check the full filename, that way you know the correct format was downloaded

Reply:
Would we want to still return the path as a string or  `pathlib.Path`? I agree with the change otherwise.

Reply:
oh yep, ideally as a `pathlib.Path`